go-controller/pkg/cluster/node.go

```diff
@@ -118,13 +103,9 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 		return err
 	}
 
-	connected, err = isOVNControllerReady(name)
-	if err != nil {
+	if _, err = isOVNControllerReady(name); err != nil {
 		return err
 	}
-	if !connected {
-		return nil
-	}
 
```

Comment:
@alexanderConstantinescu Not sure why you removed L125-127?

`isOVNControllerReady()` can return (false, nil). In that case we need to fail.

 Looks like the current code was incorrect in doing `return nil`.

Reply:
Correct me if I am wrong, but I think the current function `isOVNControllerReady` can only return `(true, nil)` (L66) or `(false, error)` (L32, L50 , L64) in the current version. And if those two are the only two return conditions, then `!connected` becomes redundant.

Reply:
If I am wrong however then we should still fix the return value for `!connected`. Returning `nil` means that we don't fail and thus do not force K8s to reboot the container. We thus don't fail but neither do we proceed to set up networking, we end up in "limbo"

Reply:
Correct me if I am wrong, but `isOVNControllerReady` can only return `(true, nil)` (L66) or `(false, error)` (L32, L50, L64). If those are our only two possible conditions then that makes `!connected` redundant.

Reply:
@alexanderConstantinescu you are right. I was thinking of something else, sorry about that.

Not this PR. Looks like `isOVNControllerReady()` should just return `error` and we should remove the `bool`.